src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,142 @@
+                } else {
+                    throw new EmptyDescriptionException("todo");
+                }
+            case "deadline":
+                Matcher deadlineMatcher = deadline.matcher(str);
+                if (deadlineMatcher.find()) {
+                    LocalDateTime deadLine = Parser.parseDate(deadlineMatcher.group(2));
+                    return new DeadlineCommand(deadlineMatcher.group(1), deadLine);
+                } else {
+                    throw new EmptyDescriptionException("deadline");
+                }
+            case "event":
+                Matcher eventMatcher = event.matcher(str);
+                if (eventMatcher.find()) {
+                    LocalDateTime timing = Parser.parseDate(eventMatcher.group(2));
+                    return new EventCommand(eventMatcher.group(1), timing);
+                } else {
+                    throw new EmptyDescriptionException("event");
+                }
+            case "find":
+                Matcher findMatcher = find.matcher(str);
+                if (findMatcher.find()) {
+                    return new FindCommand(findMatcher.group(1));
+                } else {
+                    throw new EmptyDescriptionException("find");
+                }
+            default:
+                throw new InvalidInputException();
+            }
+        }
+    }
+
+    /**
+     * Static method that takes in a string and converts it into a LocalDateTime object
+     *
+     * @param dateString String representing a date in dd/MM/yyyy HHmm format
+     * @return Object of the LocalDateTime class
+     * @throws InvalidDateException Exception when the format of the string is invalid
+     */
+    public static LocalDateTime parseDate(String dateString) throws InvalidDateException {
+        Matcher dateMatcher = datePattern.matcher(dateString);
+        try {
+            if (dateMatcher.find()) {
+                String day = dateMatcher.group(1);
+                if (day.length() == 1) {
+                    day = "0" + day;
+                } else {
+                    assert day.length() == 2;
+                }
+                String month = dateMatcher.group(2);
+
+                if (month.length() == 1) {
+                    month = "0" + month;
+                } else {
+                    assert month.length() == 2;
+                }
+                String year = dateMatcher.group(3);
+                String time = dateMatcher.group(4);
+                LocalDateTime localDateTime =
+                        LocalDateTime.parse(
+                                String.format("%s-%s-%s %s", day, month, year, time),
+                                DateTimeFormatter.ofPattern("dd-MM-yyyy HHmm"));
+                return localDateTime;
+            } else {
+                throw new InvalidDateException();
+            }
+        } catch (Exception e) {
+            throw new InvalidDateException();
+        }
```

Comment:
Are there any specific exceptions other than `InvalidDateException` that you are expecting to be thrown from the `try` block?

Reply:
No, this method will only throw an invalid date exception when there is an error in parsing the string given.